develop/github/docker-compose.yml

```diff
@@ -51,3 +51,47 @@ services:
       - discovery.type=single-node
       - xpack.security.enabled=false
       - ES_JAVA_OPTS=-Xms1g -Xmx1g
+
+  otel-collector:
+    image: otel/opentelemetry-collector-contrib:0.127.0
+    ports:
+      - "4317:4317"
+    volumes:
+      - ${PWD}/.testotel:/otel:rw
```

Comment:
I was using `.testoutput` here at first, but that lead to `open /home/runner/work/temporal/temporal/.testoutput/coverage.5731785d62f8747b1ebb0a01b785d7b4.out_0.cover.out: permission denied` errors by the test runner. I tried a few things but decided to use a separate dir instead.